Add prices to search suggestions

ravishanker opened the pull request:
Updated search suggestions with pricing

Fixes #15315 

<table><tr><td><img width=320 src="https://user-images.githubusercontent.com/990349/134106774-8bd92550-6e88-4698-9f53-d8c11fa20dbb.png"</td><td><img width=320 src="https://user-images.githubusercontent.com/990349/134106810-fc05157c-866a-473a-849d-910b9c99a8d3.png"</td></tr></table>

To test:

Wait for this [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15305) merge before testing this for better interface

1. Enable AppSettings -> Debug settings -> SiteDomainsFeatureConfig ✅
2. Return to My Site, Notice there's a new item under Configuration -> 🌐 Domains
3. Click on Domains, takes your Site Domains Screen
4. Click on Search for a domain.  Notice the domain search suggestions now show prices
5. Test with a free plan and paid plan to to see the above screens


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Add prices to search suggestions

Updated search suggestions with pricing

renanferrari commented on WordPress/src/main/java/org/wordpress/android/ui/domains/DomainSuggestionsViewHolder.kt:
```diff
@@ -33,4 +39,34 @@ class DomainSuggestionsViewHolder(
+        suggestion.is_free -> {
+            suggestion.cost
+        }
+        SiteUtils.onFreePlan(site) -> {
```
We probably shouldn't use `SiteUtils.onFreePlan` to determine if the domain is going to be free or not, because it's possible for a user on a paid plan to not have any domain credits left, in which case, they are still able to purchase a domain that is not going to be free.

Assuming I'm not missing anything, we should probably reuse some of the logic in the [`DomainRegistrationHandler`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/domainregistration/DomainRegistrationHandler.kt) class to determine if the user has domain credits or not.

ravishanker replied:
Thanks for a quick review @renanferrari.  Updated to use [`DomainRegistrationHandler`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/domainregistration/DomainRegistrationHandler.kt)  class

renanferrari reviewed (commented):
Thanks for the changes! Everything seems to be working as expected and the code is looking good. Just left one comment below.

I also think we are going to need a design review (cc @kyleaparker)

kyleaparker commented:
I reviewed just the pricing piece and it's looking good, just a couple of adjustments needed: 
- The striked out price should use the secondary text colour so that it's slightly lighter 
- It looks like the strike extends a little past the end of the word **year**, can we end it right after the **r**?
- I don't think we need to use bold for the pricing text, can we try regular? 
- We are showing free domains mixed in within this list, is that intended?


<img width=320 src="https://user-images.githubusercontent.com/712873/134098595-5f509dd7-e2ff-4c15-b7ef-c110ff528f26.png"/>

I have some more feedback not directly related to the prices, should this be left in future PR's?

ravishanker committed:
Update font styles

Updated price styles after design review

ravishanker commented:
> I reviewed just the pricing piece and it's looking good, just a couple of adjustments needed:
> 
>     * The striked out price should use the secondary text colour so that it's slightly lighter
> 
>     * It looks like the strike extends a little past the end of the word **year**, can we end it right after the **r**?
> 
>     * I don't think we need to use bold for the pricing text, can we try regular?
> 
>     * We are showing free domains mixed in within this list, is that intended?
> 
> I have some more feedback not directly related to the prices, should this be left in future PR's?

Thanks @kyleaparker for a quick review.  I've updated the three font styles of the price from the above. Please see the screen shots at the top.
  
At the moment, it is using existing domain search.  Once the entire flow is working, we can fine tune it.  You'll also have then more opportunities to do comprehensive review and tweak it to satisfaction.

ravishanker committed:
Update logic

Updated to use DomainRegistrationHandler and DomainCreditAvailable to decide for free or paid plans

ravishanker committed:
Update DomainSuggestionsViewModelTest.kt

fix test

renanferrari reviewed (approved):
@ravishanker Thanks for the changes and again, thanks for clarifying where this stands! Like with #15305, I'm merging this now so we can follow up with the adjustments it needs soon.
